src/finiteVolume/cellInterpolant.cpp

```diff
@@ -450,11 +454,25 @@ void ablate::finiteVolume::CellInterpolant::ComputeFieldGradients(const domain::
                 DMPlexApplyLimiter_Internal(dm, dmCell, lim, dim, dof, cell, field.id, cellFaces[f], faceRange.start, faceRange.end, cellPhi, xLocalArray, cellGeometryArray, cg, cx, cgrad) >>
                     utilities::PetscUtilities::checkError;
             }
+
             /* Apply limiter to gradient */
+            PetscBool cancel = PETSC_FALSE;
+            for (PetscInt pd = 0; pd < dof; ++pd) {
+                if (cellPhi[pd] == 0) {
+                    for (PetscInt d = 0; d < dim; d++) {
+                        // limit the maxium gradient, this is required for strong shocks in rocket simulations
```

Comment:
Probably expand on this comment, i.e. change it to something similar to
 // Due to directional limiting being difficult in unstructured grids, a strict gradient limiter is introduced here to revert back to cell centered reconstructions if a component limiter is 0 even though there is a strong component gradient in a direction (Usually happens at strong shocks seen in high pressured rocket simulations)